opencog/rule-engine/forwardchainer/DefaultForwardChainerCB.cc

```diff
@@ -165,21 +165,22 @@ HandleSeq DefaultForwardChainerCB::unify(Handle source,Rule* rule){
  */
 HandleSeq DefaultForwardChainerCB::subatom_unify(Handle source,Rule* rule)
 {
-    HandleSeq derived_rules;
+    HandleSeq derived_rules={};
     HandleSeq impl_members = rule->get_implicant_seq();
 
-    UnorderedHandleSet output_expanded;
     for (Handle h : impl_members) {
         UnorderedHandleSet hs = get_all_unique_atoms(h);
         hs.erase(h); //Already tried to unify this.
 
-        HandleSeq result = unify(source, rule);
-        derived_rules.insert(derived_rules.begin(), result.begin(),
-                             result.end());
+        for (Handle h : hs) {
```

Comment:
Shouldn't get_all_unique_atoms return unique list of variables? and the unit test also asserts this assumption of mine. Am I missing sth here?